server/channels/api4/channel.go

```diff
@@ -425,18 +425,9 @@ func restoreChannel(c *Context, w http.ResponseWriter, r *http.Request) {
 }
 
 func createDirectChannel(c *Context, w http.ResponseWriter, r *http.Request) {
-	userIds, err := model.NonSortedArrayFromJSON(r.Body, *c.App.Config().ServiceSettings.MaximumPayloadSizeBytes)
-	if err != nil {
-		c.Err = model.NewAppError("createDirectChannel", model.PayloadParseError, nil, "", http.StatusBadRequest).Wrap(err)
-		return
-	}
+	userIds := model.ArrayFromJSON(r.Body)
 	allowed := false
 
-	// single userId allowed if creating a self-channel
-	// NonSortedArrayFromJSON will remove duplicates, so need to add back
-	if len(userIds) == 1 && userIds[0] == c.AppContext.Session().UserId {
-		userIds = append(userIds, userIds[0])
-	}
```

Comment:
Why was this removed? The `ArrayFromJSON` calls `RemoveDuplicateStringsNonSort` so it also removes any duplicates like the current implementation does.